### PR TITLE
MAINT: make import of `array_api_compat` nicer

### PR DESCRIPTION
**Pull request by lucascolley**
#### Reference issue
Towards gh-18866

#### What does this implement/fix?
- Changes the import from e.g. `from scipy._lib.array_api_compat.array_api_compat import numpy` to just `from scipy._lib.array_api_compat import numpy`
- Uses the new import

#### Additional information
@rgommers is there a nicer way to do this? See https://github.com/mesonbuild/meson/issues/12601 for why I've used `install_data`.

- Also moves from `np.testing.assert_` to `assert` in the `xp` assertions since we're cleaning up imports

### CODE REVIEW

File: scipy/_lib/meson.build [hunk 1]
```diff
@@ -145,10 +145,31 @@ python_sources = [
...
+  'array_api_compat/torch/__init__.py',
+  'array_api_compat/torch/_aliases.py',
+  'array_api_compat/torch/linalg.py',
+]
+
+install_data(
   python_sources,
-  preserve_path: true,
-  subdir: 'scipy/_lib'
+  install_dir: py3.get_install_dir() / 'scipy/_lib',
```
Comment (lucascolley) [inline scipy/_lib/meson.build:171]:
```suggestion
  install_dir: py3.get_install_dir() / 'scipy/_lib/array_api_compat',
```

Just realised that we can do this.

**Review by rgommers — COMMENTED** [review]
Thanks Lucas, his looks like an improvement. I am not sure that this remapping works well with editable installs, but when I wanted to check that I found that we're not testing editable installs in CI and there's another problem with them preventing me from testing this. So let me fix that first, and then revisit this.

The `rename` bit indeed looks a bit clumsy, will need to look at the linked Meson issue.

**Comment by rgommers** [comment]
Editable installs seem happy with this.

> @rgommers is there a nicer way to do this?

How about using multiple separate `py3.install_sources` calls instead? Like so:
```meson
py3.install_sources(
  [
      'array_api_compat/array_api_compat/__init__.py',
      'array_api_compat/array_api_compat/_internal.py',
  ],
  subdir: 'scipy/_lib/array_api_compat',
)
py3.install_sources(
  [
    'array_api_compat/array_api_compat/common/__init__.py',
    'array_api_compat/array_api_compat/common/_aliases.py',
    'array_api_compat/array_api_compat/common/_helpers.py',
    'array_api_compat/array_api_compat/common/_linalg.py',
    'array_api_compat/array_api_compat/common/_typing.py',
  ],
  subdir: 'scipy/_lib/array_api_compat/common',
)
```

That should be shorter and a bit cleaner.

**Comment by lucascolley** [comment]
> How about using multiple separate py3.install_sources calls instead?

Ah yes, that looks good! will change

**Review by rgommers — APPROVED** [review]
LGTM now, and CI is happy. Let's give this a go - thanks Lucas!

**Pull request merged**